jira/client.py

```diff
@@ -92,7 +92,8 @@ def emit(self, record):
 #    "which means that there is a big change of having problems. " \
 #    "Possible workaround http://stackoverflow.com/a/17628350/99834" % encoding)
 
-logging.getLogger('jira').addHandler(NullHandler())
+logger = logging.getLogger('jira')
```

Comment:
> Out of curiosity I added this question on SO.

I kinda have answer to that: Loggers are composed in hierarchy, separated by dot syntax. `None` or empty string is a root logger. `'jira'` is a logger which has a root logger as parent. `'jira.utils'` would be another logger (namespace) which will have "jira" logger as a parent. By default all sub loggers propagate their messages to their parents, unless configured not to do so.

`__package__` is a good candidate, but it will create another logger namespace when used in sub packages. (Might not be always desirable.)

These two documents should have more info:  
https://docs.python.org/3/howto/logging-cookbook.html
https://docs.python.org/3/howto/logging.html